nuxeo-services/login/nuxeo-platform-login-jwt/src/test/java/org/nuxeo/ecm/jwt/TestJWTAuthenticator.java

```diff
@@ -0,0 +1,176 @@
+/*
+t * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Florent Guillaume
+ */
+package org.nuxeo.ecm.jwt;
+
+import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.impl.UserPrincipal;
+import org.nuxeo.ecm.core.api.local.ClientLoginModule;
+import org.nuxeo.ecm.core.api.local.LoginStack;
+import org.nuxeo.ecm.platform.api.login.UserIdentificationInfo;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+@RunWith(FeaturesRunner.class)
+@Features(RuntimeFeature.class)
+@Deploy("org.nuxeo.ecm.jwt")
+@Deploy("org.nuxeo.ecm.jwt.tests:OSGI-INF/test-jwt-config.xml")
+public class TestJWTAuthenticator {
+
+    protected static final String USERNAME = "bob";
+
+    protected static final String BEARER_SP = "Bearer ";
+
+    protected static final String ACCESS_TOKEN = "access_token";
+
+    @Inject
+    protected JWTService service;
+
+    @Before
+    public void setUp() throws Exception {
+        LoginStack loginStack = ClientLoginModule.getThreadLocalLogin();
+        loginStack.push(new UserPrincipal(USERNAME, Collections.emptyList(), false, false), null, null);
```

Comment:
These setUp and teardown instructions could be a junit rule. As it we'll be able to reuse it easily by just instantiating it.

Reply:
Ok I'll think of it next time although there's not much reuse to do in this case.

Reply:
Indeed, just two usages in this PR.